File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/StartStreamBuilder.java
package wpam.mobile_client.protocol;

import java.io.Closeable;
import java.util.ArrayList;

	public final class StartStreamBuilder implements CommandBuilder, Closeable
	{
		public StartStreamBuilder()
		{
		}
		public final void close()
		{
		}

		@Override
		public Command create(ArrayList<Integer> commandInBytes)
		{
			Command command  = new StartStreamCommand();
			return command;
		}
	}

File: server/src/protocol/StartStreamBuilder.cpp
#include "StartStreamBuilder.h"
#include "StartStreamCommand.h"

using namespace std;
using namespace communication;

StartStreamBuilder::StartStreamBuilder()
{}

StartStreamBuilder::~StartStreamBuilder()
{}

unique_ptr<Command> StartStreamBuilder::create(const vector<uint8_t> &commandInBytes)
{
    auto command = make_unique<StartStreamCommand>();

    return move(command);
}